src/main/java/gift/controller/AdminProductController.java

```diff
@@ -0,0 +1,92 @@
+package gift.controller;
+
+import gift.dto.ProductRequestDto;
+import gift.dto.ProductResponseDto;
+import gift.service.ProductService;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Controller
+@RequestMapping("/admin/products")
+public class AdminProductController {
+
+    private final ProductService productService;
+
+    public AdminProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    //목록 화면
+    @GetMapping
+    public String list(@RequestParam(defaultValue = "0") int page,
+                       @RequestParam(defaultValue = "10") int size,
+                       Model model) {
+
+        // 서비스 호출
+        List<ProductResponseDto> products = productService.getProductList(page, size);
+
+        // 총 페이지 계산용
+        int total = productService.getProductCount();
+        int totalPages = (int) Math.ceil((double) total / size);
```

Comment:
```suggestion
        int totalPages = (total + size - 1) / size;
```
여담으로 실수오차 피하고 높은 정밀도를 원하면 이런 방법도 좋아요~ 단, total + size 가 int 범위를 넘지 않는다면요!